File: Backend/Organizer/Models/Event.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Organizer.Models
{
    [Table("event")]
    public class Event
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Eid { get; set; }   // PK, AI

        [Required]
        [StringLength(100)]
        public string Ename { get; set; }  // NOT NULL

        [Required]
        public int Vid { get; set; }   // FK → venue.vid (no join here)

        [Required]
        public TimeSpan Time { get; set; } // TIME NOT NULL

        [Required]
        public DateTime Date { get; set; } // DATE NOT NULL

        [Required]
        [Column(TypeName = "decimal(10,2)")]
        public decimal Fair { get; set; }  // DECIMAL(10,2) NOT NULL

        [Column(TypeName = "text")]
        public string? Description { get; set; } // NULL allowed

        [Required]
        [Column(name:"uid")]
        public int OrganizerId { get; set; } // user.uid (organizer)
    }
}

File: Backend/Admin/src/main/java/com/example/demo/Entity/Event.java
package com.example.demo.Entity;

import java.sql.Date;
import java.sql.Time;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@JsonIgnoreProperties({"venue", "organizer"})

@Entity
@Table(name = "event")
public class Event {
	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long eid;

    private String ename;
    private Time time;
    private Date date;
    private double fair;
    private String description;
    /** 0 = PENDING, 1 = APPROVED, 2 = REJECTED (matches techfestemsdb.event.status int) */
    private Integer status;
    private int capacity;

    @ManyToOne(fetch = jakarta.persistence.FetchType.LAZY)
    @JoinColumn(name = "uid", nullable = false)
    private User organizer;

    @ManyToOne
    @JoinColumn(name = "vid")
    private Venue venue;

    public Integer getStatus() {
		return status;
	}

	public void setStatus(Integer status) {
		this.status = status;
	}

	public int getCapacity() {
		return capacity;
	}

	public void setCapacity(int capacity) {
		this.capacity = capacity;
	}

	public User getOrganizer() {
		return organizer;
	}

	public void setOrganizer(User organizer) {
		this.organizer = organizer;
	}

	public Long getEid() {
		return eid;
	}

	public void setEid(Long eid) {
		this.eid = eid;
	}

	public String getEname() {
		return ename;
	}

	public void setEname(String ename) {
		this.ename = ename;
	}

	public Time getTime() {
		return time;
	}

	public void setTime(Time time) {
		this.time = time;
	}

	public Date getDate() {
		return date;
	}

	public void setDate(Date date) {
		this.date = date;
	}

	public double getFair() {
		return fair;
	}

	public void setFair(double fair) {
		this.fair = fair;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public Venue getVenue() {
		return venue;
	}

	public void setVenue(Venue venue) {
		this.venue = venue;
	}
}
